### PR TITLE
Issue #50: amp: Implement Key Management Service

### PR DESCRIPTION
**Pull request by danielcahoon**

### CODE REVIEW

**Review by gstragand — APPROVED** [review]
lgtm

File: pkg/amp/key_management.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		return route.OK
+	}
+
+	// Commands that can be handled locally
+	switch req.Command() {
+	case route.Load:
+		return k.RouteInOrder(req)
+	case route.Provision:
+		if err := k.Provision(req.Flags().Get()...); err != nil {
+			return route.FAIL
```
Comment (icahoon) [inline pkg/amp/key_management.go]: You should probably to a msg.Error(err.Error()) before returning.

File: pkg/amp/encryption_key.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+		msg.Detail("Encryption Key does not exist, skipping...")
+		return nil
+	}
+	if err := k.createSecurityTags(); err != nil {
+		return err
+	}
+	return k.ProviderEncryptionKey().Provision(flags...)
+}
+func (k *encryptionKey) Audit(flags ...string) error {
+	if len(flags) == 0 || flags[0] == "" {
```
Comment (icahoon) [inline pkg/amp/encryption_key.go:161]: Do you need to create an audit session if it doesn't already exist. I am assuming that you can do: amp <acct> key foo audit, and get to this code without a session already existing.

File: pkg/amp/storage.go [hunk 3]
```diff
@@ -135,6 +135,10 @@ func (s *storage) Route(req *route.Request) route.Response {
 	case "bucket":
 		req.Pop()
 		bucket := s.FindBucket(req.Top())
+		if req.Top() == "" {
```
Comment (icahoon) [inline pkg/amp/storage.go]: This code seems placed incorrectly. Should if go before the FindBucket call?

File: pkg/aws/encryption_key.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+		EncryptionKey: cfg,
+		keyManagement: key.KeyManagement().ProviderKeyManagement().(*keyManagement),
+		kms:           prov.kms,
+	}
+	k.set(k.keyManagement.encryptionKeyCache.find(k))
+
+	return k, nil
+}
+func (k *encryptionKey) Load() error {
+	return nil
```
Comment (icahoon) [inline pkg/aws/encryption_key.go]: Does this need to be implemented? Should it be pulling a value out of a cache?

**Review by icahoon — APPROVED** [review]
lgtm

**Comment by danielcahoon** [comment]
Closes Issue #50

**Pull request merged**